### PR TITLE
[ENH] Pytorch Classifier & de-novo implementation of Transformer

### PR DESCRIPTION
**Pull request by geetu040**
#### Reference Issues/PRs

Fixes https://github.com/sktime/sktime/issues/6786

#### What does this implement/fix? Explain your changes.

- adds a pytorch based classifier base class
- implements transformer classifier
  - Paper: [A Transformer-based Framework for Multivariate Time Series Representation Learning](https://dl.acm.org/doi/abs/10.1145/3447548.3467401)
  - Code: [gzerveas/mvts_transformer](https://github.com/gzerveas/mvts_transformer)

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
very nice! Some basic API/architecture comments or the start.

* quick question, is this from scratch, or are you using someon else's code? If the latter, we need to cite those persoms.
* there should be no loose torch imports on module level, or the soft dependency is not isolated. Kindly use the pattern for other torch layers.

**Comment by geetu040** [comment]
> * quick question, is this from scratch, or are you using someon else's code? If the latter, we need to cite those persoms.

original authors have been cited in the files where code has been copied

> * there should be no loose torch imports on module level, or the soft dependency is not isolated. Kindly use the pattern for other torch layers.

changes have been made to ensure the pattern

File: sktime/classification/deep_learning/transformer.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    >>> X_train, y_train = load_unit_test(split="train")
+    >>> X_test, _ = load_unit_test(split="test")
+    >>>
+    >>> model = TransformerClassifier()
+    >>> model.fit(X_train, y_train)  # doctest: +SKIP
+    >>> preds = model.predict(X_test)  # doctest: +SKIP
+    """
+
+    _tags = {
+        "authors": ["geetu040"],
```
Comment (fkiraly) [inline sktime/classification/deep_learning/transformer.py:86]: also add original authors to the `"authors"` tag, please

Comment (geetu040) [reply sktime/classification/deep_learning/transformer.py]: added

File: sktime/classification/deep_learning/transformer.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+    from torch.utils.data import DataLoader, Dataset
+else:
+
+    class Dataset:
+        """Dummy class if torch is unavailable."""
+
+
+class TransformerClassifier(BaseDeepClassifierPytorch):
```
Comment (fkiraly) [inline sktime/classification/deep_learning/transformer.py:17]: there are multiple transformer based architectures for various tasks including classification - can we give it a more specific name, hence?

Comment (geetu040) [reply sktime/classification/deep_learning/transformer.py]: Yes, you are right, it should not be given a generic name of `transformer`. As it is not the official transformer architecture, rather a proposed transformer architecture from an academic publication. The authors themselves call it "MVTS Transformer (Multivariate Time Series Transformer)".

Therefore I have renamed
- classes from `TransformerClassifier` to `MVTSTransformerClassifier`
- files from `transformer` to `mvts_transformer`

For reference: 
  - Paper: [A Transformer-based Framework for Multivariate Time Series Representation Learning](https://dl.acm.org/doi/abs/10.1145/3447548.3467401)
  - Code: [gzerveas/mvts_transformer](https://github.com/gzerveas/mvts_transformer)

**Review by fkiraly — APPROVED** [review]
LGTM, great addition!

merged from main and updated author tag

**Pull request merged**